[boost] update to 1.76.0

autoantwort opened the pull request:
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17330

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  lets test, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Still a draft

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

autoantwort commented:
The problem with boost math and boost nowide is that they are recursive. E.g.
```
[1/2] cd /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build && /mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/b2 --user-config=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/user-config.jam --stagedir=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/stage --build-dir=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel runtime-link=static link=static address-model=64 architecture=x86 target-os=linux toolset=gcc threadapi=pthread -sZLIB_BINARY=z -sZLIB_LIBPATH="/mnt/vcpkg-ci/installed/x64-linux/lib/libz.a" -sBZIP2_BINARY=bz2 -sBZIP2_LIBPATH="/mnt/vcpkg-ci/installed/x64-linux/lib/libbz2.a" variant=release --layout=system --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -sZLIB_INCLUDE="/mnt/vcpkg-ci/installed/x64-linux/include" -sBZIP2_INCLUDE="/mnt/vcpkg-ci/installed/x64-linux/include" -sICU_PATH="/mnt/vcpkg-ci/installed/x64-linux" -j16 -sBOOST_ROOT=/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build -sBOOST_BUILD_PATH=/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build --debug-configuration --ignore-site-config --hash -q threading=multi debug-symbols=on stage
FAILED: CMakeFiles/boost 
cd /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build && /mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/b2 --user-config=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/user-config.jam --stagedir=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/stage --build-dir=/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel runtime-link=static link=static address-model=64 architecture=x86 target-os=linux toolset=gcc threadapi=pthread -sZLIB_BINARY=z -sZLIB_LIBPATH="/mnt/vcpkg-ci/installed/x64-linux/lib/libz.a" -sBZIP2_BINARY=bz2 -sBZIP2_LIBPATH="/mnt/vcpkg-ci/installed/x64-linux/lib/libbz2.a" variant=release --layout=system --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -sZLIB_INCLUDE="/mnt/vcpkg-ci/installed/x64-linux/include" -sBZIP2_INCLUDE="/mnt/vcpkg-ci/installed/x64-linux/include" -sICU_PATH="/mnt/vcpkg-ci/installed/x64-linux" -j16 -sBOOST_ROOT=/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build -sBOOST_BUILD_PATH=/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build --debug-configuration --ignore-site-config --hash -q threading=multi debug-symbols=on stage
notice: found boost-build.jam at /mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/boost-build.jam
notice: loading B2 from /mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/bootstrap.jam
notice: Site configuration files will be ignored due to the
notice: --ignore-site-config command-line option.
notice: Loading explicitly specified user configuration file:
    /mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/user-config.jam
notice: Searching '/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from '/mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel'.
notice: will use '/usr/bin/c++' for gcc, condition <toolset>gcc-5.4.1
notice: using gcc libraries :: <toolset>gcc-5.4.1 :: /usr/bin /usr/lib /usr/lib32 /usr/lib64
notice: using gcc archiver :: <toolset>gcc-5.4.1 :: /usr/bin/ar
notice: using gcc ranlib :: <toolset>gcc-5.4.1 :: /usr/bin/ranlib
warning: toolset gcc initialization: can not find tool windres
warning: initialized from /mnt/vcpkg-ci/buildtrees/boost-nowide/x64-linux-rel/user-config.jam:22
notice: using rc compiler :: <toolset>gcc-5.4.1 :: /usr/bin/as
Performing configuration checks

    - std::fstream is moveable and swappable : yes [1]
    - Has Large File Support   : yes [1]
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:609: in start-building from module targets
error: Recursion in main target references
error: the following target are being built currently:
error: ../build/stage /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build/config/checks/cxx11_defaulted_functions /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build/config/checks/cxx11_noexcept /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build/config/checks/cxx11_rvalue_references /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build/config/checks/cxx11_static_assert /mnt/vcpkg-ci/buildtrees/boost-nowide/src/ost-1.76.0-c006322863.clean/build/config/checks/cxx11_defaulted_functions
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:776: in class@main-target.generate from module object(main-target)@1458
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:934: in targets.generate-from-reference from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:355: in builds-raw from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:457: in configure.builds from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:512: in check from module object(check-target-builds-worker)@192
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:107: in modules.call-in from module object(check-target-builds-worker)@192
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/util/indirect.jam:105: in indirect.call from module indirect
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/property.jam:132: in property.evaluate-conditionals-in-context from module property
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1088: in evaluate-requirements from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1122: in common-properties2 from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1018: in targets.common-properties from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1314: in class@basic-target.generate from module object(typed-target)@934
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:812: in generate-really from module object(main-target)@1506
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:784: in class@main-target.generate from module object(main-target)@1506
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:934: in targets.generate-from-reference from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:355: in builds-raw from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:457: in configure.builds from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:512: in check from module object(check-target-builds-worker)@195
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:107: in modules.call-in from module object(check-target-builds-worker)@195
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/util/indirect.jam:105: in indirect.call from module indirect
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/property.jam:132: in property.evaluate-conditionals-in-context from module property
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1088: in evaluate-requirements from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1122: in common-properties2 from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1018: in targets.common-properties from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1314: in class@basic-target.generate from module object(typed-target)@914
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:812: in generate-really from module object(main-target)@1501
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:784: in class@main-target.generate from module object(main-target)@1501
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:934: in targets.generate-from-reference from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:355: in builds-raw from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:457: in configure.builds from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:512: in check from module object(check-target-builds-worker)@194
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:107: in modules.call-in from module object(check-target-builds-worker)@194
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/util/indirect.jam:105: in indirect.call from module indirect
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/property.jam:132: in property.evaluate-conditionals-in-context from module property
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1088: in evaluate-requirements from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1122: in common-properties2 from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1018: in targets.common-properties from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1314: in class@basic-target.generate from module object(typed-target)@866
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:812: in generate-really from module object(main-target)@1489
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:784: in class@main-target.generate from module object(main-target)@1489
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:934: in targets.generate-from-reference from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:355: in builds-raw from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:457: in configure.builds from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:512: in check from module object(check-target-builds-worker)@193
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:107: in modules.call-in from module object(check-target-builds-worker)@193
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/util/indirect.jam:105: in indirect.call from module indirect
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/property.jam:132: in property.evaluate-conditionals-in-context from module property
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1088: in evaluate-requirements from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1122: in common-properties2 from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1018: in targets.common-properties from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1314: in class@basic-target.generate from module object(typed-target)@742
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:812: in generate-really from module object(main-target)@1458
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:784: in class@main-target.generate from module object(main-target)@1458
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:934: in targets.generate-from-reference from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:355: in builds-raw from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:457: in configure.builds from module configure
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/configure.jam:512: in check from module object(check-target-builds-worker)@192
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:107: in modules.call-in from module object(check-target-builds-worker)@192
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/util/indirect.jam:105: in indirect.call from module indirect
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/property.jam:132: in property.evaluate-conditionals-in-context from module property
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1088: in evaluate-requirements from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1122: in common-properties2 from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1018: in targets.common-properties from module targets
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:1314: in class@basic-target.generate from module object(install-target-class)@208
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:812: in generate-really from module object(main-target)@213
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build/targets.jam:784: in class@main-target.generate from module object(main-target)@213
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/build-system.jam:797: in load from module build-system
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/modules.jam:295: in import from module modules
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/src/kernel/bootstrap.jam:139: in boost-build from module
/mnt/vcpkg-ci/installed/x64-linux/tools/boost-build/boost-build.jam:8: in module scope from module

ninja: build stopped: subcommand failed.
```
But I don't know jam and don't know how to fix that :/

autoantwort commented:
> The problem with boost math and boost nowide is that they are recursive. 

I have created boostorg/build#724

FabienPean commented:
It would be nice to have a fix for #17882 with this new version

autoantwort committed:
[boost] update generator script for boost 1.76

autoantwort committed:
[boost] update ports to 1.76.0 (run generator)

autoantwort committed:
[boost] fix windows build?

autoantwort committed:
[quantlib] update and fix mac build

autoantwort commented:
Would also fix #17705

autoantwort committed:
[symengine] update and fix build

autoantwort committed:
[avro-cpp] update to latest master and fix windows build

autoantwort committed:
[folly] update to 2021.05.31.00

autoantwort commented:
Now the pipeline fails for thinks that worked a few commits ago on mac and linux, and I only changed windows things 😕

autoantwort committed:
[fbthrift, fizz, wangle] update to v2021.05.31.00 and fix build

autoantwort commented:
@PhoebeHui Can you help me with the windows build. The reported failure is `fatal error C1083: Cannot open include file: 'zstd.h': No such file or directory` even when the CMakeLists.txt contains 
```cmake
find_package(ZLIB REQUIRED)
target_link_libraries(... ZLIB::ZLIB ....)
```

dg0yt commented:
> The reported failure is `fatal error C1083: Cannot open include file: 'zstd.h': No such file or directory` even when the CMakeLists.txt contains
> 
> ```cmake
> find_package(ZLIB REQUIRED)
> target_include_directories(... ${ZLIB_INCLUDE_DIRS} ....)
> ```

`find_package(ZLIB)` doesn't match the desired  'zstd.h'. Do dependencies match?

autoantwort commented:
> `find_package(ZLIB)` doesn't match the desired 'zstd.h'. Do dependencies match?

Ah you mean `zstd.h` is from the dependency `zstd` and not `zlib`? Currently it only haves a dependency to `zlib`.

autoantwort committed:
[proxygen] update to version 2021.05.31.00

autoantwort commented:
I have no further idea how to fix facebook/proxygen#366, so this PR depends in this issue.

autoantwort committed:
[fizz, proxygen, fbthrift] fix sodium target

autoantwort committed:
[proxygen] also works on macOS

autoantwort commented:
@PhoebeHui You can review the PR, if it looks good to you, I will run `vcpkg x-add-version -all` (I don't want to run it every time when I have to change a lot)

PhoebeHui commented on ports/fizz/portfile.cmake:
```diff
@@ -3,11 +3,9 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+    SHA512 529fa0e32cb0aecf59fb90f1d8c9bf284c6749169713c365ca06bbf54ed12a8978042af05094214c42945d372e2457bb2bfc2acc543f6538b296a25eb66d7186
     HEAD_REF master
-    PATCHES
-        find-zlib.patch
```
It looks fizz depends on zlib, however it doesn't link to the project, so this patch may still be needed. And zlib should be added to fizz dependency.

autoantwort replied:
I have created facebookincubator/fizz#61 for that. Imho this should also be fixed upstream

autoantwort replied:
But this fix will be in the release next Monday

autoantwort replied:
Seems that they have broken their releases. I will add a patch.

autoantwort replied:
@PhoebeHui The problem is when I apply the patch in fizz I get the following error in fbthrift:
```
CMake Error at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake:596 (_add_library):
  Target "async" links to target "ZLIB::ZLIB" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  thrift/lib/cpp/CMakeLists.txt:85 (add_library)


CMake Error at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake:596 (_add_library):
  Target "thriftcpp2" links to target "ZLIB::ZLIB" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  thrift/lib/cpp2/CMakeLists.txt:127 (add_library)


CMake Error at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake:596 (_add_library):
  Target "thriftprotocol" links to target "ZLIB::ZLIB" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  thrift/lib/cpp2/CMakeLists.txt:91 (add_library)


CMake Error at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/scripts/buildsystems/vcpkg.cmake:596 (_add_library):
  Target "thriftfrozen2" links to target "ZLIB::ZLIB" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  thrift/lib/cpp2/CMakeLists.txt:51 (add_library)
```
Any idea?

autoantwort replied:
But there is a `find_package(Zlib REQUIRED)` as far I can see

PhoebeHui replied:
I will try to repro this in my local VM.

autoantwort replied:
@PhoebeHui Idk I use x64-osx and have no x64-linux

autoantwort replied:
I think the `rsocket` dependency is completely unnecessary and can be removed

PhoebeHui replied:
I checked fbthrift:x64-osx, it passed in CI testing, I will confirm this in my local VM.

autoantwort replied:
> fbthrift:x64-osx and fbthrift:x64-linux built successfully
> It due to try to find zlib via thrift\cmake\FindZlib.cmake, remove it will solve the issue

I have done that, now `fbthrift` fails for x64-linux with 
```
CMake Error at /agent/_work/2/s/scripts/buildsystems/vcpkg.cmake:764 (_find_package):
  By not providing "FindZlib.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Zlib", but
  CMake did not find one.

  Could not find a package configuration file provided by "Zlib" with any of
  the following names:

    ZlibConfig.cmake
    zlib-config.cmake

  Add the installation prefix of "Zlib" to CMAKE_PREFIX_PATH or set
  "Zlib_DIR" to a directory containing one of the above files.  If "Zlib"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:109 (find_package)
```

autoantwort replied:
Ok seems to be fixed upstream in https://github.com/facebook/fbthrift/commit/4034aeb18854f274b8c3207485346fd27e06fba0

autoantwort replied:
For me fbthrift fails on windows with
```
...
-- Found BISON: C:/Users/LeanderSchulten/git_projects/Lichtsteuerung.git/vcpkg/downloads/tools/winflexbison/0a14154bff-a8cf65db07/win_bison.exe (found suitable version "3.1", minimum required is "3.0.4") 
CMake Error at C:/Users/LeanderSchulten/git_projects/Lichtsteuerung.git/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find FLEX (missing: FLEX_EXECUTABLE)
```
But there is a flex executable right next to the bison executable 😕

autoantwort replied:
> After solve the error, it starts to fail with 'LINK : fatal error LNK1104: cannot open file 'lib\compiler_ast.lib'', maybe a source issue, @autoantwort, could you please have a look?

I have no idea, I have created https://github.com/facebook/fbthrift/issues/433  
But it works for `x64-windows-static`, currently `fbthrift` does not build at all on windows. So this PR should not be blocked by this.

PhoebeHui replied:
@autoantwort, please update the baseline version, I think it's unrelate to these changes. We can open seperate PR to fix it when the Upstream has updates.

PhoebeHui reviewed (changes requested):
@autoantwort, appreciate your effort very much!

autoantwort commented on ports/symengine/fix-build.patch:
```diff
@@ -0,0 +1,23 @@
+From ceae675dccd593f119f3010d46e133de74fc01bc Mon Sep 17 00:00:00 2001
+From: Liam Keegan <liam@keegan.ch>
+Date: Mon, 31 May 2021 21:31:02 +0200
+Subject: [PATCH] use list initialization for int parameter since type depends
+ on boost version
+
+---
+ symengine/mp_boost.cpp | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
```
In 5 days I can update the version and this fix is not needed anymore

autoantwort replied:
It was another library that got a new release on Monday, so I have minimized the patch file and fixed deprecation warnings

autoantwort commented on ports/quantlib/portfile.cmake:
```diff
@@ -27,6 +27,7 @@ vcpkg_configure_cmake(
     OPTIONS
         -DUSE_BOOST_DYNAMIC_LIBRARIES=${USE_BOOST_DYNAMIC_LIBRARIES}
         -DMSVC_RUNTIME=${QL_MSVC_RUNTIME}
+        -DCMAKE_CXX_STANDARD=11
```
Only on mac. Maybe I simply wait until lballabio/QuantLib#1108 gets merged

PhoebeHui replied:
The PR has been merged, but it will need some time to release, so I think we can add a patch to solve this firstly, and remove it in next revision.

PhoebeHui commented:
Features test:
- [X] boost-regex[icu]:x64-windows
- [ ] boost-locale[icu]:x64-windows
- [ ] avro-cpp[*]:x64-windows
- [ ] folly[*]:x64-windows
- [ ] symengine[*]:x64-windows
- [ ] boost-regex[icu]:x64-linux
- [ ] boost-locale[icu]:x64-linux
- [ ] avro-cpp[*]:x64-linux
- [ ] folly[*]:x64-linux
- [ ] symengine[*]:x64-winux

Need double check if these ports passed in CI testing.
 - [ ] avro-cpp
 - [ ] wangle
 - [ ] symengine
 - [ ] proxygen
 - [ ] folly
 - [ ] fizz
 - [ ] fbthrift

autoantwort committed:
[quantlib] use fix from upstream to fix mac build

autoantwort committed:
[symengine] minimize patch file and fix deprecation warning

autoantwort committed:
[folly,proxygen,wangle,fizz,fbthrift] update to 2021.06.14.00

autoantwort committed:
[fbthrift] remove unnecessary dependency rsocket

I couldn't find any information that this dependency exists. The term is used in the code, but not in the context of a dependency

autoantwort committed:
[fizz,fbthrift] fix zlib dependency

autoantwort committed:
[fbthrift] pass required flex executable to cmake configure

PhoebeHui commented:
The features passed with x64-windows and x64-linux.
fbthrift:x64-windows failures has Uptream issue https://github.com/facebook/fbthrift/issues/433 to track, it's not relate to these changes, It should not block this PR.

@ras0219-msft @ras0219, could you please help review?

autoantwort committed:
add version files

Ryan-rsm-McKenzie commented on scripts/boost/generate-ports.ps1:
```diff
@@ -323,12 +312,15 @@ foreach ($library in $libraries)
             $groups = $(
                 findstr /si /C:"include <boost/" include/*
                 findstr /si /C:"include <boost/" src/*
+                findstr /si /C:"include \`"boost/" include/*
+                findstr /si /C:"include \`"boost/" src/*
```
You could avoid the duplicate calls by using the `/r` switch to enable regular expressions:
```suggestion
                findstr /rsi /C:"include [<\`"]boost/" include/*
                findstr /rsi /C:"include [<\`"]boost/" src/*
```

PhoebeHui commented:
@autoantwort, could you address the review suggestions?

autoantwort committed:
[boost] generate-ports.ps1: Apply code review

autoantwort committed:
[boost] changes from new version of generate-ports script

autoantwort committed:
update version files

autoantwort commented:
@Ryan-rsm-McKenzie Thanks for your review. It also fixed some dependencies 👍

strega-nil-ms commented on scripts/boost/generate-ports.ps1:
```diff
@@ -321,14 +310,14 @@ foreach ($library in $libraries)
         if ($IsWindows)
         {
             $groups = $(
-                findstr /si /C:"include <boost/" include/*
-                findstr /si /C:"include <boost/" src/*
+                findstr /rsi /C:"include [<\`"]boost/" include/*
+                findstr /rsi /C:"include [<\`"]boost/" src/*
             ) | % { $_ -replace "^[^:]*:","" }
         }
         else
         {
             $groups = $(
-                grep -irhs "include <boost/" include src
+                grep -irhs "include [<\`"]boost/" include src
             )
         }
```
```suggestion
        Get-ChildItem -Recurse -LiteralPath include,src | % {
            Get-Content $_ | where { $_ -match 'include [<"]boost/' }
        }
```

just use powershell stuff here.

strega-nil-ms reviewed (approved):
just minor change requested

strega-nil-ms commented:
Here are my CRs in a patch file: [0001-Nicole-CRs.patch.txt](https://github.com/microsoft/vcpkg/files/6772642/0001-Nicole-CRs.patch.txt)

autoantwort committed:
[boost] generate-ports.ps1: Apply code review

autoantwort committed:
Merge branch 'master' into boost-1.76.0

# Conflicts:
#	ports/boost-modular-build-helper/vcpkg.json
#	versions/b-/boost-modular-build-helper.json
#	versions/baseline.json

strega-nil-ms commented:
@vicroms could you merge this today?
